### PR TITLE
chore: authenticate with npm before publishing package changes

### PR DESCRIPTION
**Pull request by martin3walker**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -112,10 +109,9 @@ jobs:
       - yarn_install
       - vault/get-secrets:
           template-preset: semantic-release-ecosystem
-      # Lerna only looks for the name GH_TOKEN
```
Comment (martin3walker) [inline .circleci/config.yml:115]: This should be already done in vault.

**Review by andipaetzold — APPROVED** [review]
Let's wait for tolkien's approval to make sure we didn't mess things up

**Review by 2wce — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 2]
```diff
@@ -28,14 +28,11 @@ cache-key: &cache-key
 commands:
   with_npm_token:
```
Comment (chrishelgert) [inline .circleci/config.yml:29]: As it is now only used within the release step, can we remove it and use the command directly?

**Review by chrishelgert — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (denkristoffer) [reply .circleci/config.yml:115]: You need the `vault/configure-lerna` step

File: .circleci/config.yml [hunk 3]
```diff
@@ -112,10 +105,8 @@ jobs:
       - yarn_install
       - vault/get-secrets:
           template-preset: semantic-release-ecosystem
-      # Lerna only looks for the name GH_TOKEN
-      - run: export GH_TOKEN=${GITHUB_TOKEN}
-      - run: git config --global user.email "${GIT_AUTHOR_EMAIL}"
-      - run: git config --global user.name "${GIT_AUTHOR_NAME}"
+      - vault/configure-lerna
+      - run: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> ~/.npmrc
```
Comment (martin3walker) [inline .circleci/config.yml:109]: @chrishelgert does this look alright?

File: .circleci/config.yml [hunk 4]
```diff
@@ -112,10 +105,8 @@ jobs:
       - yarn_install
       - vault/get-secrets:
           template-preset: semantic-release-ecosystem
-      # Lerna only looks for the name GH_TOKEN
-      - run: export GH_TOKEN=${GITHUB_TOKEN}
-      - run: git config --global user.email "${GIT_AUTHOR_EMAIL}"
-      - run: git config --global user.name "${GIT_AUTHOR_NAME}"
+      - vault/configure-lerna
```
Comment (martin3walker) [inline .circleci/config.yml:108]: @denkristoffer Is this correct?

**Review by chrishelgert — APPROVED** [review]

**Pull request merged**